csharp/extractor/Semmle.Extraction.CSharp/Entities/Compilations/Compilation.cs

```diff
@@ -39,45 +39,29 @@ public override void Populate(TextWriter trapFile)
             trapFile.compilation_assembly(this, assembly);
 
             // Arguments
-            var index = 0;
-            foreach (var arg in Compilation.Settings.Args)
-            {
-                trapFile.compilation_args(this, index++, arg);
-            }
+            Compilation.Settings.Args.ForEach((arg, index) => trapFile.compilation_args(this, index, arg));
 
             // Files
-            index = 0;
-            foreach (var file in Context.Compilation.SyntaxTrees.Select(tree => File.Create(Context, tree.FilePath)))
-            {
-                trapFile.compilation_compiling_files(this, index++, file);
-            }
+            Context.Compilation.SyntaxTrees.Select(tree => File.Create(Context, tree.FilePath)).ForEach((file, index) => trapFile.compilation_compiling_files(this, index, file));
```

Comment:
Would be better to have another `ForEach` overload that doesn't include index, and use that instead of `Select`.

Reply:
Actually, since `Select` is lazily evaluated, I believe one would have to add something like `.ToArray()` at the end.

Reply:
What about 
```csharp
Context.Compilation.SyntaxTrees.ForEach((tree, index) => trapFile.compilation_compiling_files(this, index, File.Create(Context, tree.FilePath)));
```

Reply:
Sorry, I misread; I thought the `ForEach` call happened inside the lambda passed to `Select`, and not on the result of `Select`. All should be good then.